client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationBuilder.java

```diff
@@ -635,8 +635,9 @@ public Configuration create() {
             throw new IllegalArgumentException("Both marshaller and marshallerClass attributes are present, but marshaller is not an instance of marshallerClass");
       }
 
-      Map<String, RemoteCacheConfiguration> remoteCaches = remoteCacheBuilders.entrySet().stream().collect(Collectors.toMap(
-            Map.Entry::getKey, e -> e.getValue().create()));
+      // Collect to a concurrent map to handle concurrent access for read/writes.
+      Map<String, RemoteCacheConfiguration> remoteCaches = remoteCacheBuilders.entrySet().stream()
+            .collect(Collectors.toConcurrentMap(Map.Entry::getKey, e -> e.getValue().create()));
```

Comment:
If we use a ConcurrentMap are any of the changes really required?

Reply:
I've done that because the `addRemoteCache` uses the synchronized block. But looking closer, I can revert all places and change only the `addRemoteCache` to use the compute method instead.